src/sf-mcp-server.ts

```diff
@@ -41,15 +42,21 @@ export class SfMcpServer extends McpServer implements ToolMethodSignatures {
   /** Optional telemetry instance for tracking server events */
   private telemetry?: Telemetry;
 
+  private dynamicTools: boolean;
+
   /**
    * Creates a new SfMcpServer instance
    *
    * @param {Implementation} serverInfo - The server implementation details
    * @param {ServerOptions & { telemetry?: Telemetry }} [options] - Optional server configuration including telemetry
    */
-  public constructor(serverInfo: Implementation, options?: ServerOptions & { telemetry?: Telemetry }) {
+  public constructor(
+    serverInfo: Implementation,
+    options?: ServerOptions & { telemetry?: Telemetry; dynamicTools?: boolean }
```

Comment:
Why aren't `telemetry` and `dynamicTools` part of the `ServerOptions` type?  Also, this changes the jsDoc to be inconsistent.

Reply:
ServerOptions is taken from the mcp sdk - so I'm merging our stuff with theirs